### ISSUE TITLE
API calls stopped working on Humio Cloud

### ISSUE THREAD

**Issue by wknapik** [humio/issues#114]
Hi,

We're using Humio Cloud v1.11.0. The automation we've had in place for quite a while, for creating views and adding users to them, started failing with errors like:
```
{"data":null,"errors":[{"message":"Cannot query field 'addMember' on type 'Mutation'. Did you mean 'addUser'? (line 1, column 12):\nmutation { addMember(searchDomainName:\"redacted\", username:\"redacted\",hasDeletionRights:false, hasMembershipAdminRights:false) { clientMutationId } }\n           ^","locations":[{"column":12,"line":1}]}]}
```
We have not touched this code in a long time, so it seems like a change on the Humio side.

I just started investigating this, so I don't have more information at the moment, but I figured, I'd open this issue in case this is something obvious to you guys. Have there been any backwards-incompatible changes to the API recently ?

Also, I've tried replacing our existing code with calls to humioctl, which would be much more readable/maintainable, but unfortunately, I instantly ran into https://github.com/humio/cli/issues/26 ;/

**Comment by andejens** [comment]
thank you - we will take a look

**Comment by JacobJust** [comment]
That api endpoint is replaced with:

mutation {
  updateDefaultGroupMemberships(input: {userName:"user", viewName:"viewname", groups:[Member, Admin, Eliminator]}) {
    groups {
      users {
        username
      }
    }
  }
}

**Comment by andejens** [comment]
Hi @wknapik - please let us know if it works for you.

**Comment by wknapik** [comment]
Hi @andejens & @JacobJust,

Yes, this does work, so thanks for the snippet.

The bigger problem here is that an unversioned API is changing in a backwards-incompatible way.

I have a few questions about that:
* If the API is going to be versioned in the future
  * Will we be notified of a new version becoming available ?
  * How many versions will be maintained at any given time ?
  * How much time will we have before old versions are no longer supported ? Will we be notified when (or before) they reach EOL ?
* if the API is NOT going to be versioned in the future
  * Will we be notified of backwards-incompatible changes ahead of time ?
  * How much time will we have to adapt ?
  * How will we be able to adapt to API changes without versioning ? Will there be a separate API instance with the changes, available at a different url ?

This was unexpected and quite disruptive for us. We'd like to make sure it doesn't happen again.

---

PS. I've also noticed an extreme slowdown in the processing of api requests. A call to updateDefaultGroupMemberships can take up to 1min 20s to process. What's up with that ? Is that some sort of rate-limiting scheme ? If so, it kicks in very quickly.

**Comment by wknapik** [comment]
Hi @andejens. Any updates on this ?

**Comment by wknapik** [comment]
Hi @andejens. It's been 3 weeks. Do you have an update ?

**Comment by andejens** [comment]
Yeah - sorry about that - for now we are not having any active plans for versioning the API, we are evaluating this going forward. My suggestion would be to set up a meeting to discuss this in more detail. I'll reach out to you by mail and coordinate a meeting.

**Comment by wknapik** [comment]
Hi. Our API calls started failing again today. Have there been more backwards-incompatible changes recently ? View creation is failing.

**Comment by wknapik** [comment]
Hello ?

**Comment by wknapik** [comment]
@andejens / @JacobJust ?

**Comment by wknapik** [comment]
Can we please get an answer ? It's been 6 days now.

**Comment by wknapik** [comment]
@andejens, @JacobJust it's been 3 weeks, can you please respond ?
